lib/db/schema/patch-022-023.sql

```diff
@@ -0,0 +1,133 @@
+-- This migration:
+--
+-- * Changes the type of the devices.callbackPublicKey column
+-- * Add a new devices.callbackAuthKey column
+-- * Updates the accountDevices stored procedure with the new column
+-- * Alters three stored procedures, createDevice, updateDevice and
+--   sessionWithDevice to work with the new columns
+
+ALTER TABLE devices
+MODIFY COLUMN callbackPublicKey VARCHAR(88),
+ADD COLUMN callbackAuthKey VARCHAR(24);
```

Comment:
According to [1], changing the datatype of a column blocks concurrent queries to that column, so deploying this might imply a service outage if the devices table is large.  We may be able to get away with a two-step process:

 * A first `ALTER TABLE` that renames `callbackPublicKey` to `oldCallbackPublicKey` and creates a new `callbackPublicKey` with the required datatype.
* A second `ALTER TABLE` that drops the `oldCallbackPublicKey` column

Which IIUC would take longer, but only contains operations that allow concurrent queries on the table.  This also has the advantage of removing any soon-to-be-junk values that may have made their way into the existing `callbackPublicKey` column.

[1] https://dev.mysql.com/doc/refman/5.6/en/innodb-create-index-overview.html#innodb-online-ddl-summary-grid